doc/api/process.md

```diff
@@ -2305,6 +2305,222 @@ import { loadEnvFile } from 'node:process';
 loadEnvFile();
 ```
 
+### Environment variables file parser specification
+
+This section describes how the environment variables file parser reads a file
+and sets up the environment variables in Node.js.
```

Comment:
Add something about the source and how it's different like: 

> While `.env` files descend from shell scripts that exported environment variables there are a few important distinctions in how they work regarding quoting, spacing and escaping values.


It may also make sense to mention the dotenv packge which popularized the format